src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -157,7 +159,13 @@ public void teleportHome(@NotNull Player player, @NotNull Island island, int del
      */
     private void teleportHome(@NotNull Player player, @NotNull Island island) {
         player.setFallDistance(0);
-        player.teleport(LocationUtils.getSafeLocation(island.getHome(), island));
+        Location homeSafeLocation = LocationUtils.getSafeLocation(island.getHome(), island);
+        if (!PaperLib.isPaper()) {
```

Comment:
This isnt needed PaperLib.teleportAsync will fall back onto player.teleport if you arnt running paper

Reply:
ye i said this too, but das says that method has bugs

Reply:
Not necessarily bugs, but it should be tested

Reply:
Personally, I had done the tests with basic spigot and it worked without problems for me

Reply:
https://youtu.be/ecZOMX9Pcik

Reply:
lol u can say "i tested" don't needed a video <3

Reply:
I preferred to give proof of what I said anyway :)